ingester/service/submissions_tracker.go

```diff
@@ -0,0 +1,162 @@
+package service
+
+import (
+	"database/sql"
+
+	"chainlink/ingester/client"
+	"chainlink/ingester/logger"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+
+	"go.uber.org/multierr"
+
+	"fmt"
+	"sync"
+	"time"
+)
+
+// SubmissionsTracker is an interface for subscribing to new Chainlink aggregator feeds
+type SubmissionsTracker interface {
+	Start() error
+	Stop()
+}
+
+type submissionsTracker struct {
+	db           *sql.DB
+	eth          client.ETH
+	mutex        sync.Mutex
+	subs         map[string]client.Subscription
+	feedsTracker FeedsTracker
+}
+
+// NewSubmissionsTracker
+func NewSubmissionsTracker(db *sql.DB, eth client.ETH, feedsTracker FeedsTracker) SubmissionsTracker {
+	ht := &submissionsTracker{
+		db:           db,
+		eth:          eth,
+		feedsTracker: feedsTracker,
+		mutex:        sync.Mutex{},
+		subs:         map[string]client.Subscription{},
+	}
+	return ht
+}
+
+// Start subscribes to new submissions when a new feed is added
+func (st *submissionsTracker) Start() error {
+	newFeeds := st.feedsTracker.Subscribe()
+
+	go func() {
+		for {
+			select {
+			case aggregator, open := <-newFeeds:
+				if !open {
+					return
+				}
+				go st.listenForSubmissions(aggregator)
+			}
+		}
+	}()
+
+	return nil
+}
+
+// Stop unsubscribes from all submissions for the tracked feeds
+func (st *submissionsTracker) Stop() {
+	st.mutex.Lock()
+	defer st.mutex.Unlock()
+
+	for _, sub := range st.subs {
```

Comment:
Generally I think a better way to go about doing this is to signal the "worker" loop, and have the worker loop perform shut down, this also gives it a chance to gracefully finish any work it is in the middle of and reduces the chances of race conditions.

Then you'd typically have a wait group that you release to allow `Stop` to unblock.